packages/fx-core/src/core/middleware/utils/MigrationUtils.ts

```diff
@@ -289,6 +290,10 @@ function provisionOutputNamingConverterV3(
             break;
           }
         } while ((outputNames = pluginRegex.exec(bicepContent)));
+      } else if ("fx-resource-bot" === pluginNames[0] && keyName.endsWith("AppResourceId")) {
+        // keep AppResourceId aligned in fx-resource-bot plugin
+        outputName = outputNames[1];
+        keyName = "functionAppResourceId";
```

Comment:
can you add some comments to explain why we need to change the key name here?

Reply:
First of all, I want to make sure that whether we need to change the key name to keep the placeholder of azure function resource id the same(PROVISIONOUTPUT__BOTOUTPUT__FUNCTIONAPPRESOURCEID).
What's your opinion?

Reply:
Now I changed the logic in app yml generating